hashes/src/internal_macros.rs

```diff
@@ -21,14 +68,32 @@
 /// `internal_engine` is required to initialize the engine for given hash type.
 macro_rules! hash_trait_impls {
     ($bits:expr, $reversed:expr $(, $gen:ident: $gent:ident)*) => {
```

Comment:
Unrelated but at some stage it would be nice to use a single term, currently we have `$reversed` and `$reverse` - I think we should favour `$reverse` because it matches `DISPLAY_BACKWARD` (i.e., `$reversed` pairs with `DISPLAYED_BACKWARD`).